ta4j-core/src/main/java/org/ta4j/core/Decimal.java

```diff
@@ -72,24 +74,28 @@ private Decimal(String val) {
         delegate = new BigDecimal(val, MATH_CONTEXT);
     }
 
-    /**
-     * Constructor.
-     * @param val the double value
-     */
-    private Decimal(double val) {
-        delegate = BigDecimal.valueOf(val);
+    private Decimal(short val) {
+        delegate = new BigDecimal(val, MATH_CONTEXT);
     }
 
     private Decimal(int val) {
-        delegate = BigDecimal.valueOf(val);
+        delegate = new BigDecimal(val, MATH_CONTEXT);
     }
 
     private Decimal(long val) {
         delegate = BigDecimal.valueOf(val);
     }
 
```

Comment:
In  #84 the creation of Decimal based on `double`, `int` and `long` was changed to `valueOf(int | long )` to increase performance and accuracy. Why do you change back to `Decimal` constructor for int?